mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -935,53 +935,81 @@ LAGenericRewritePattern::matchAndRewrite(linalg::GenericOp laGeneric,
   return success();
 }
 
-LogicalResult
-MemcpyRewritePattern::matchAndRewrite(memref::CopyOp copy,
-                                      LinalgAlignRewriter &b) const {
+template <typename WriteOp>
+static LogicalResult
+rewriteMemCpy(memref::CopyOp copy, LinalgAlignRewriter &b,
+              function_ref<void(WriteOp, Value)> destUpdateFunc) {
   auto src = copy.getSource();
   auto target = copy.getTarget();
   Location loc = copy.getLoc();
 
+  // See if the copy source is transformed.
+  // If so we need to reverse them on the destination.
+  auto [rawSrc, transformsOnCpySrc, isBig] = untransform(b, src);
+  std::ignore = isBig;
+
   Operation *gemmStoreOp = nullptr;
   SmallVector<TransformMapAttr> views;
-  if (auto allocOp = src.getDefiningOp<memref::AllocOp>()) {
-    if (auto twop = dyn_cast_if_present<ThreadwiseWriteAllOp>(
-            traceToWriter(src, views))) {
+  LLVM_DEBUG(llvm::dbgs() << "copy source = " << rawSrc << "\n");
+  if (auto allocOp = rawSrc.getDefiningOp<memref::AllocOp>()) {
```

Comment:
"No transforms on readers" is an invariant of the output fusion chain.

If `rawSrc != source`, then you need to go up to `Regularize.cpp` and fix it

Reply:
I m just not going to do that now.
So I dont accept this as a reason to block this PR merely because this make this pass more robust if the situation arises.

However, you are more than welcome to fail/handle in regularization.

Reply:
> "the generic/copy/[thing that gets fused away] does not take a view of the intermediate alloc"

This is an artificial restriction that would otherwise could be solved here.